bitcoin-magazine/bitcoin-explained/explaining-rgb-tokens-the-technical-side-of-bitcoin.md

```diff
@@ -0,0 +1,1254 @@
+And so first, the USD Tether people would have to issue this token.
+And generally, the way that's done in RGB is actually you do use an OP_RETURN.
+So first, there's going to be a transaction.
+The transaction has an OP_RETURN, so everybody can see it.
+And that transaction says, hereby, I declare that now inside of this new UTXO there are just 1 million dollars worth of USD Tether.
+
+Sjors Provoost: 00:21:52
+
+Yeah, and this is actually where your hash at least comes up, because what is in the OP_RETURN as far as I know is a hash of a JSON file.
+So anybody who has the JSON file will know that the hash refers to it.
+And they can then read the JSON file and see what the supply is and what the rules are for whatever asset has been created.
+
+Aaron van Wirdum: 00:22:14
+
+A JSON file is just text, right?
+
+Sjors Provoost: 00:22:17
+
+It's just text.
+Yeah.
+With annoying brackets.
+
+Ruben Somsen: 00:22:20
+
+So, okay, so now we have a UTXO with $1 million in their USD Tether dollars IOUs.
+And when they first send them to a buyer, somebody who says, "Okay, I'll give you $1,000, you give me $1,000 worth of these coins."
+So the first thing that...
+
+## Client-Side Validation
+
+Ruben Somsen: 00:22:37
+
+...RGB does, is client-side validation, meaning that the JSON file that Sjors just pointed out needs to be transferred during the coin transfer.
+So as soon as these coins move from one owner to the next owner, you also have to provide all the data, which refers to the hash that you put in the Bitcoin blockchain.
+
+Aaron van Wirdum: 00:22:55
+
+Right.
+
+Sjors Provoost: 00:22:56
+
+So the first person who buys the tether, basically, has to receive an email.
+And inside that email should be the JSON file, and I guess also a proof that they now own the coins.
+
+Aaron van Wirdum: 00:23:09
+
+Right.
+So that's the big difference compared to colored coins, a counterparty, is that now we have a separate layer..
+...essentially, of JSON file texts that are also being sent around and they're being linked to in the Bitcoin blockchain.
+Is that right?
+
+Sjors Provoost: 00:23:24
+
+Yeah, and this also detaches the amount that's really being moved from the amount of Satoshis that's being moved.
+So yes, you're making a transaction.
+
+Aaron van Wirdum: 00:23:33
+
+But I think that was already the case for counterparty.
+
+Sjors Provoost: 00:23:35
+
+Yeah, for counterparty it is, but for colored coins it wasn't.
+For colored coins, the Satoshis really mattered.
+But in the case of RGB, you're just sending somebody a transaction and then along with the transaction, an email with what's actually happening.
+And that transaction might move one Satoshi or a thousand Satoshis, doesn't really matter.
+The instructions are separate, and are not on the chain.
+Right.
+
+Ruben Somsen: 00:23:54
+
+So, okay, so we have this one output with $1 million worth of USD Tether, and now they want to send $1,000 to somebody else.
```

Comment:
Another example where "So, okay, so" can be removed from the start of the sentence.